web/client-api/src/main/java/io/deephaven/web/client/api/widget/JsWidgetExportedObject.java

```diff
@@ -0,0 +1,47 @@
+package io.deephaven.web.client.api.widget;
+
+import elemental2.promise.Promise;
+import io.deephaven.javascript.proto.dhinternal.io.deephaven.proto.table_pb.ExportedTableCreationResponse;
+import io.deephaven.javascript.proto.dhinternal.io.deephaven.proto.ticket_pb.TypedTicket;
+import io.deephaven.web.client.api.Callbacks;
+import io.deephaven.web.client.api.JsTable;
+import io.deephaven.web.client.api.WorkerConnection;
+import io.deephaven.web.client.state.ClientTableState;
+import jsinterop.annotations.JsMethod;
+import jsinterop.annotations.JsProperty;
+
+public class JsWidgetExportedObject {
+    private final WorkerConnection connection;
+
+    private final TypedTicket ticket;
+
+    public JsWidgetExportedObject(WorkerConnection connection, TypedTicket ticket) {
+        this.connection = connection;
+        this.ticket = ticket;
+    }
+
+    @JsProperty
+    String getType() {
+        return ticket.getType();
+    }
+
+    @JsMethod
+    Promise<Object> fetch() {
+        if (!getType().equals("Table")) {
+            // TODO (deephaven-core#62) implement fetch for tablemaps
```

Comment:
seems like non-tables could be fetched _in this case_ as objects, just call connection.getObject(...)? We have the full typed ticket, we can make the var def (or make a util method on WorkerConnection to handle an existing typed ticket).

Reply:
Not sure we could? We don't have the name or id of the object I don't think?

Reply:
I'm dumb, should be good now.